File: 백준/Bronze/1546. 평균/평균.py
import sys
input = sys.stdin.read

def re_avg(data: list[str]):
    n = int(data[0])
    nlist = list(map(int, data[1:]))
    max_val = max(nlist)
    
    print(sum(nlist) / max_val * 100 / n)

def solve():
    data = input().split()
    if not data: return
    
    re_avg(data)

if __name__ == "__main__":
    solve()

File: 백준/Bronze/1546. 평균/평균.rs
use std::io::{self, Read};

fn main() {
    let mut input = String::new();
    io::stdin().read_to_string(&mut input).unwrap();
    let mut tokens = input.split_whitespace();

    let n: usize = tokens.next().unwrap().parse().unwrap();

    let mut scores = Vec::with_capacity(n);
    let mut max_val: f64 = 0.0;

    for _ in 0..n {
        let val: f64 = tokens.next().unwrap().parse().unwrap();
        if val > max_val {
            max_val = val;
        }
        scores.push(val);
    }

    let sum: f64 = scores.iter()
        .map(|&s| (s / max_val) * 100.0)
        .sum();

    println!("{}", sum / n as f64);
}